### PR TITLE
fix(repos): fix namings after salt binaries and keys move

### PR DESCRIPTION
**Pull request by sticky-note**
BREAKING CHANGES: This commit changes upstream repository naming conventions to get latest binaries and keys. If you depends on versions of salt from older, please stick with version 1.x.x of this formula

Fixup of some requested changes from https://github.com/saltstack-formulas/salt-formula/pull/548
as we need this to work in order to unbreak formula for new versions of Salt

### PR progress checklist (to be filled in by reviewers)

- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [x] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

Yes

### Documentation checklist

- [x] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

**Comment by lelutin** [comment]
Thanks @sticky-note for tackling the problems with the move of the repo key. The current code on the master branch is not working properly because of this.

I don't feel qualified to add feedback on the code changes as a whole though since there are more changes than what I would currently use.

File: salt/osfamilymap.yaml [hunk 1]
```diff
@@ -25,9 +29,9 @@
 
 
 Debian:
-  pkgrepo: 'deb [signed-by=/usr/share/keyrings/salt-archive-keyring.gpg arch=amd64] {{ salt_repo }}/{{ py_ver_repr or 'py3' }}/{{ osfamily_lower }}/{{ osmajorrelease }}/amd64/{{ salt_release }} {{ oscodename }} main'
-  pkgrepo_keyring: '{{ salt_repo }}/{{ py_ver_repr or 'py3' }}/{{ osfamily_lower }}/{{ osmajorrelease }}/amd64/{{ salt_release }}/salt-archive-keyring.gpg'
-  pkgrepo_keyring_hash: sha256=ea38e0cdbd8dc53e1af154a8d711a2a321a69f81188062dc5cde9d54df2b8c47
+  pkgrepo: 'deb [signed-by=/usr/share/keyrings/salt-archive-keyring.gpg arch={{ repoarch }}] {{ salt_repo }}/{{ py_ver_repr or 'py3' }}/{{ osfamily_lower }}/{{ osmajorrelease }}/{{ repoarch }}/{{ salt_release }} {{ oscodename }} main'
+  pkgrepo_keyring: '{{ salt_repo }}/{{ py_ver_repr or 'py3' }}/{{ osfamily_lower }}/{{ osmajorrelease }}/amd64/{{ salt_release }}/SALT-PROJECT-GPG-PUBKEY-2023.gpg'
```
Comment (primechuck) [inline salt/osfamilymap.yaml:33]: New PGP url also has arch in path.  Should update to use the new osarch.

**Comment by primechuck** [comment]
I was starting a PR for this, but saw yours which is much more inclusive.  Any additional changes for this so we don't have to override pillar data to get salt to deploy salt?

**Comment by sticky-note** [comment]
Obsoleted by #566 . Can be closed

**Pull request closed without merging**